File: erpnext_facturama/facturacionorcom/page/vista_almacen/vista_almacen.py
import frappe

@frappe.whitelist()
def get_stock_items(warehouse=None, search=None, item_group=None):
    filters = {}
    if warehouse:
        filters["warehouse"] = warehouse

    conditions = ""
    if search:
        conditions += " AND (i.name LIKE %(search)s OR i.item_name LIKE %(search)s)"
    if item_group:
        conditions += " AND i.item_group = %(item_group)s"

    data = frappe.db.sql(
        f"""
        SELECT
            i.name AS identificador,
            i.item_name AS nombre,
            COALESCE(i.description, '') AS descripcion,
            COALESCE(i.item_group, '') AS grupo,
            COALESCE(SUM(b.actual_qty), 0) AS stock,
            COALESCE(b.warehouse, '') AS almacen
        FROM `tabItem` i
        LEFT JOIN `tabBin` b ON b.item_code = i.name
            {("AND b.warehouse = %(warehouse)s" if warehouse else "")}
        WHERE i.disabled = 0
            {conditions}
        GROUP BY i.name, b.warehouse
        ORDER BY i.item_group, i.item_name
        """,
        {
            "warehouse": warehouse,
            "search": f"%{search}%" if search else None,
            "item_group": item_group,
        },
        as_dict=True,
    )
    return data


@frappe.whitelist()
def get_warehouses():
    return frappe.get_all("Warehouse", filters={"is_group": 0}, fields=["name"], order_by="name")

File: erpnext_facturama/facturacionorcom/page/vista_almacen/vista_almacen.js
frappe.pages["vista-almacen"].on_page_load = function (wrapper) {
	const page = frappe.ui.make_app_page({
		parent: wrapper,
		title: "Vista Almacén",
		single_column: true,
	});

	// Filtro de almacén
	const warehouse_field = page.add_field({
		fieldtype: "Link",
		fieldname: "warehouse",
		options: "Warehouse",
		label: "Almacén",
		change() {
			load_data();
		},
	});

	// Filtro de grupo de producto
	const item_group_field = page.add_field({
		fieldtype: "Link",
		fieldname: "item_group",
		options: "Item Group",
		label: "Grupo de Producto",
		change() {
			load_data();
		},
	});

	// Buscador
	const search_field = page.add_field({
		fieldtype: "Data",
		fieldname: "search",
		label: "Buscar",
		change() {
			load_data();
		},
	});

	page.add_button(__("Actualizar"), () => load_data(), { icon: "refresh" });

	// Contenedor de la tabla
	const $body = $(`
		<div class="vista-almacen-wrapper" style="padding: 15px;">
			<div class="table-responsive">
				<table class="table table-bordered table-hover">
					<thead class="thead-dark">
						<tr>
							<th>Identificador</th>
							<th>Nombre</th>
							<th>Grupo</th>
							<th>Descripción</th>
							<th style="text-align:right;">Stock</th>
							<th>Almacén</th>
						</tr>
					</thead>
					<tbody id="almacen-tbody">
						<tr><td colspan="6" class="text-center text-muted">Cargando...</td></tr>
					</tbody>
				</table>
			</div>
		</div>
	`).appendTo(page.main);

	function load_data() {
		const warehouse = warehouse_field.get_value();
		const search = search_field.get_value();
		const item_group = item_group_field.get_value();

		frappe.call({
			method: "erpnext_facturama.facturacionorcom.page.vista_almacen.vista_almacen.get_stock_items",
			args: { warehouse, search, item_group },
			callback(r) {
				const rows = r.message || [];
				const $tbody = $("#almacen-tbody");
				$tbody.empty();

				if (!rows.length) {
					$tbody.html(
						'<tr><td colspan="6" class="text-center text-muted">Sin resultados</td></tr>'
					);
					return;
				}

				rows.forEach((row) => {
					$tbody.append(`
						<tr>
							<td><a href="/app/item/${row.identificador}">${row.identificador}</a></td>
							<td>${row.nombre}</td>
							<td>${row.grupo || ""}</td>
							<td>${(row.descripcion || "").replace(/<[^>]*>/g, "").trim()}</td>
							<td style="text-align:right;">${frappe.format(row.stock, { fieldtype: "Float" })}</td>
							<td>${row.almacen}</td>
						</tr>
					`);
				});
			},
		});
	}

	load_data();
};
